### ISSUE TITLE
Search function can't find the test "CLINITESTR"

### ISSUE THREAD

**Issue by Trunkenstein** [zerforschung/schnelltesttest.de#19]
The "CLINITEST® Rapid COVID-19 Antigen Self-Test", produced by Healgen, distributed by Siemens can't be found with the Keyowrds "Siemens" or "CLINITEST".
![found](https://user-images.githubusercontent.com/5569258/150305731-ae8afba1-d72d-4d62-81c8-68afa27fbb3f.PNG)
![search](https://user-images.githubusercontent.com/5569258/150305733-6502ee5c-ac8d-4e89-bd35-ffc161bf1c27.PNG)
![siemens](https://user-images.githubusercontent.com/5569258/150305736-1f837811-dfd6-4413-a974-bc5b2dff0bd9.PNG)

Also, the EAN (10816490025075) for the "20x Test Kit", used in schools, is missing.
![ClinitestR](https://user-images.githubusercontent.com/5569258/150306649-36a52b43-ff63-4b5c-b3d8-d54e21e545fd.png)

Tried on Android with latest Bromite and Chrome as well as Windows with Iron Chrome

**Comment by tpulatha** [comment]
If you send a picture with the Ref number (most likely AT001/20) and the EAN I can open a pull request.

**Comment by cobber** [comment]
From what I can see (`src/data/all.csv`, `src/routes/EnterCode.tsx` and `src/utils/testdata.ts`), the data for my schnelltests (`Anhui Deepblue...`) are there, but the search only checks prefixes: so `deep` or `blue` doesn't turn up any useful results.

The barcode on the box is also way too small for me to read and even when I enter the number it doesn't match.

Would it be better to use a case-insensitive regexp.
e.g.:
1. split user input into words
2. find all items which match any words in the item's data, ignoring case
3. sort by decreasing order of number of words found, then alphabetically

Proof of concept:
(Note: I've never done any typescript, this is just plain JS, tested on FF 96.0.1 and safari 14.1.2)

```
data       = [ `a`, `b`, `ab`, `c b a`, `a b`, `b a` ]
user_input = `b c`
user_words = user_input.split(/\s+/)
match_re   = RegExp( `(${user_words.join(`|`)})`, 'gi' )
ratings    = data.map( n => [ Array.from( n.matchAll(match_re) ).length, n ] )
hits       = ratings.filter( m => m[0] > 0 )
best_hits  = hits.sort( (a,b) => ( b[0] - a[0] ) || a[1].localeCompare( b[1] ) )
result     = best_hits.map( m => m[1] )
```

output:
```
    "c b a"
    "a b"
    "b"
    "b a"
```

**Comment by pajowu** [comment]
The checkprefix is only used if you enter a code that is not known to the database. We do fuzzy searching using https://fusejs.io/, but there still seems to be some issues

**Comment by cobber** [comment]
in that case, 👍 if you can improve it :-)

**Comment by so-langsam** [comment]
Searching for "Siemens", "CLINITEST" or the EAN ("10816490025075") shows the correct test now.

Maybe this was solved by #63, too.

**Comment by pajowu** [comment]
Seems like it was solved by it \o/
